### PR TITLE
[SPARK-34282][SQL][TESTS] Unify v1 and v2 TRUNCATE TABLE tests

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
1. Move parser tests from `DDLParserSuite` to `TruncateTableParserSuite`.
2. Port DS v1 tests from `DDLSuite` and other test suites to `v1.TruncateTableSuiteBase` and to `v1.TruncateTableSuite`.
3. Add a test for DSv2 `TRUNCATE TABLE` to `v2.TruncateTableSuite`.

### Why are the changes needed?
To improve test coverage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running new test suites:
```
$ build/sbt -Phive-2.3 -Phive-thriftserver "test:testOnly *TruncateTableSuite"
$ build/sbt -Phive-2.3 -Phive-thriftserver "test:testOnly *CatalogedDDLSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @HyukjinKwon Please, review this PR.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/TruncateTableSuite.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.execution.command
+
+/**
+ * The class contains tests for the `TRUNCATE TABLE` command to check V2 table catalogs.
+ */
+class TruncateTableSuite extends command.TruncateTableSuiteBase with CommandSuiteBase {
+
+  // TODO(SPARK-34290): Support v2 TRUNCATE TABLE
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/TruncateTableSuite.scala:28]: The v2 API is ready, and this should be easy to do?

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/TruncateTableSuite.scala:28]: Not yet, this https://github.com/apache/spark/pull/31420 is the first step.

**Review by cloud-fan — APPROVED** [review]
LGTM. Can you fix the conflicts?

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**